catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java

```diff
@@ -73,4 +79,42 @@ public static Integer convertToInteger(Tre tre, String fieldName) {
         }
         return Optional.empty();
     }
+
+    public static Float convertToFloat(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            return Float.valueOf((String) value);
+        } else {
+            return null;
+        }
+    }
+
+    public static Boolean convertYnToBoolean(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            String valueStr = (String) value;
+            return valueStr.equalsIgnoreCase("Y");
+        } else {
+            return null;
+        }
+    }
+
+    public static Date convertToDate(Tre tre, String fieldName) {
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            String dateStr = (String) value;
+            SimpleDateFormat sdf = new SimpleDateFormat(TRE_DATE_FORMAT);
```

Comment:
We should consider using https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/time/FastDateFormat.html instead as it doesn't require a new instance every time.